FRW-1241: Adjusted strict search documentation (#1558)
spryker/spryker-docs: docs/scos/dev/back-end-development/zed-ui-tables/create-and-configure-zed-tables.md

    \Spryker\Zed\Gui\Communication\Table\TableConfiguration::SORT_DESC);
```

### Configure search by columns

The default search option in Back Office data tables searches for anything that contains the specified substrings. This default search makes use of the SQL logical operator ‘LIKE’ in combination with ‘LOWER’ for comparison. It may result in performance issues on larger tables due to indexes not being used.

> Search by columns can be used on all backoffice data tables which extend the `AbstractTable` class.

Reviewer: ```suggestion
Search by columns can be used on all Back Office data tables which extend the `AbstractTable` class.
```

Searchable fields are `case sensitive` and use exact comparisons by concrete column.

It is possible to enable it on a per-table basis, by setting `setSearchableColumns` on the table configuration.
The search fields appear under each column that was set as searchable.
This provides flexibility to enable/disable search fields only for needed columns:

```php
<?php

$config->setSearchableColumns([
    static::COL_ID_SALES_ORDER => SpySalesOrderTableMap::COL_ID_SALES_ORDER,
    static::COL_FIRST_NAME => SpySalesOrderTableMap::COL_FIRST_NAME,
    static::COL_EMAIL => SpySalesOrderTableMap::COL_EMAIL,
    static::COL_ORDER_REFERENCE => SpySalesOrderTableMap::COL_ORDER_REFERENCE,
]);
```

Search by columns initiates once searchable columns are enabled and the searchable terms are passed.
In case two or more searchable columns are filled the searchable condition is getting stricter 
with the `AND` operator and searching for the results via all those fields.

We recommend using search by columns for tables with large numbers of rows. 
Measurable performance checks show the execution time got faster. 

> For example, getting the results using search by columns for tables with ~200k rows 

Reviewer: ```suggestion
For example, getting the results using search by columns for tables with ~200k rows improved speed from 0,031s to 0,003s. 

```

> increased from 0,031s to 0,003s. 

Reviewer: ```suggestion
```

> ### Configure strict search

Reviewer 2: ```suggestion
### Configure search by columns
```

Please update the documentation below, as we would have just added a search by columns, not a `strict search`.
Please make sure, that this paragraph only refers to the existing search for comparison reasons.

The default search option in Back Office data tables searches for anything that contains the specified substrings. This default search makes use of the SQL logical operator ‘LIKE’ in combination with ‘LOWER’ for comparison. It may result in performance issues on larger tables due to indexes not being used.

> In order to solve the performance issues, a strict search mode has been introduced and can be used on all backoffice data tables which extend the `AbstractTable` class.

Reviewer 2: > In order to solve the performance issues

We've already mentioned performance improvements below. (Also, this is not the only reason why we need to search per column)

Please get rid of this message.

It is possible to enable it on a per-table basis, by setting `setSearchableColumns` on the table configuration.
The search fields appear under each column that was set as searchable.
This provides flexibility to enable/disable strict search fields only for needed columns:

```php
<?php

$config->setSearchableColumns([
    static::COL_ID_SALES_ORDER => SpySalesOrderTableMap::COL_ID_SALES_ORDER,
    static::COL_FIRST_NAME => SpySalesOrderTableMap::COL_FIRST_NAME,
    static::COL_EMAIL => SpySalesOrderTableMap::COL_EMAIL,
    static::COL_ORDER_REFERENCE => SpySalesOrderTableMap::COL_ORDER_REFERENCE,
]);
```

Strict search inits once searchable columns are enabled and the searchable terms are passed.
In case two or more searchable columns are filled the searchable condition is getting stricter 
with the `AND` operator and searching for the results via all those fields.

We recommend using a strict search for tables with large numbers of rows. 
Measurable performance checks show the execution time got faster. 
For example, getting the results using a strict search for tables with ~200k rows 

> increased from 6400ms to 3900ms. 

Reviewer 2: Is it query time, or the page load?

Author: Page load execution time

Reviewer 2: Please update to query execution time.
